### PR TITLE
update the translation of `promise.finally()`

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

update the translation of `promise.finally()`
### Motivation

update the translation of `promise.finally()`
### Additional details

none

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 1]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
 ### 参数
 
 - `onFinally`
-  - : `Promise` 结束后调用的 {{jsxref("Function")}}。
+  - : 一个当 promise 得到解决时异步执行的函数。它的返回值将被忽略，除非返回的值是被拒绝的 promise。调用该函数时不带任何参数。
 
 ### 返回值
 
-返回一个设置了 `finally` 回调函数的 {{jsxref("Promise")}} 对象。
+返回等效的 {{jsxref("Promise")}}。如果处理程序抛出错误或返回被拒绝的 promise，那么由 `finally()` 返回的 promise 将被该值拒绝。否则，处理程序的返回值不会影响原始 promise 的状态。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:27]:
```suggestion
返回等效的 {{jsxref("Promise")}}。如果处理程序抛出错误或返回被拒绝的 promise，那么 `finally()` 返回的 promise 将以该值被拒绝。否则，处理程序的返回值不会影响原始 promise 的状态。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 2]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
 
 ## 描述
 
-如果你想在 promise 执行完毕后无论其结果怎样都做一些处理或清理时，`finally()` 方法可能是有用的。
+如果你想在 promise 解决时进行一些处理或者清理，而不管结果如何，那么 `finally()` 方法会很有用。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:31]:
```suggestion
如果你想在 promise 敲定时进行一些处理或者清理，无论其结果如何，那么 `finally()` 方法会很有用。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 3]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+
+`finally()` 方法类似于调用 {{jsxref("Promise/then", "then(onFinally, onFinally)")}}。然而，有几个不同之处：
+
+- 在内联中创建函数时，可以传递一次，而不是强制声明两次或为其创建变量。
+- `onFinally` 回调未接收任何参数。这个用例正好适用于*不关心*拒绝原因或实现价值的情况，因此无需提供它。
+- `finally()` 调用通常是易懂的，不会更改原始 promise 的状态。例如：
+  - 与 `Promise.resolve(2).then(() => 77, () => {})` 不同，它最后返回一个值为 `77` 的 promise，`Promise.resolve(2).finally(() => 77)` 最后返回一个值为 `2` 的 promise。
+  - 同样，与 `Promise.reject(3).then(() => {}, () => 88)` 不同，它最后返回一个值为 `88` 的 promise，`Promise.reject(3).finally(() => 88)` 返回一个被拒绝的 promise，理由是 `3`。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:39]:
```suggestion
- 创建内联函数时，你可以将其传入一次，而不是强制声明两次或为其创建变量。
- `onFinally` 回调函数不接收任何参数。这种情况恰好适用于你*不关心*拒绝原因或兑现值的情况，因此无需提供它。
- `finally()` 调用通常是透明的，不会更改原始 promise 的状态。例如：
  - 与 `Promise.resolve(2).then(() => 77, () => {})` 不同，它返回一个最终会兑现为值 `77` 的 promise，而 `Promise.resolve(2).finally(() => 77)` 返回一个最终兑现为值 `2` 的 promise。
  - 类似地，与 `Promise.reject(3).then(() => {}, () => 88)` 不同，它返回一个最终兑现为值 `88` 的 promise，而 `Promise.reject(3).finally(() => 88)` 返回一个最终以原因 `3` 拒绝的 promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 4]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+
+> **备注：** `finally` 回调中的 `throw`（或返回被拒绝的 promise）仍然拒绝返回的 promise。例如，`Promise.reject(3).finally(() => { throw 99; })` 和 `Promise.reject(3).finally(() => Promise.reject(99))` 都以理由 `99` 拒绝返回的 promise。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:41]:
```suggestion
> **备注：** 在 `finally` 回调函数中抛出错误（或返回被拒绝的 promise）仍然拒绝返回的 promise。例如，`Promise.reject(3).finally(() => { throw 99; })` 和 `Promise.reject(3).finally(() => Promise.reject(99))` 都以理由 `99` 拒绝返回的 promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 5]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
 
-`finally()` 虽然与 `.then(onFinally, onFinally)` 类似，它们不同的是：
+与 {{jsxref("Promise/catch", "catch()")}} 一样，`finally()` 在内部调用其调用对象上的 `then` 方法。如果 `onFinally` 不是函数，则调用 `then()` 时同时使用 `onFinally` 作为两个参数——对于 {{jsxref("Promise.prototype.then()")}}，这意味着没有附加有用的处理程序。否则，使用两个内部创建的函数调用 `then()`，其行为如下：
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:43]:
```suggestion
与 {{jsxref("Promise/catch", "catch()")}} 一样，`finally()` 在内部调用其调用对象上的 `then` 方法。如果 `onFinally` 不是函数，则调用 `then()` 时使用 `onFinally` 同时作为两个参数——对于 {{jsxref("Promise.prototype.then()")}}，这意味着没有附加有效的处理程序。否则，`then()` 被调用时会使用两个内部创建的函数，其行为如下：
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 6]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+promise.then(
+  (value) => Promise.resolve(onFinally()).then(() => value),
+  (reason) =>
+    Promise.resolve(onFinally()).then(() => {
+      throw reason;
+    }),
+);
+```
+
+因为 `finally()` 调用 `then()`，所以它支持子类化。此外，请注意上面的 {{jsxref("Promise.resolve()")}} 调用——实际上，`onFinally()` 的返回值是使用与 `Promise.resolve()` 相同的算法解析的，但用于构造解析的 promise 的实际构造函数将是子类。`finally()` 通过 [`promise.constructor[@@species]`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/@@species) 获取构造函数。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:57]:
```suggestion
因为 `finally()` 调用 `then()`，所以它支持子类化。此外，请注意上面的 {{jsxref("Promise.resolve()")}} 调用——实际上，`onFinally()` 的返回值是使用与 `Promise.resolve()` 相同的算法解决的，但用于构造解决的 promise 的实际构造函数将是子类。`finally()` 通过 [`promise.constructor[@@species]`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/@@species) 获取构造函数。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 7]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+    if (contentType && contentType.includes("application/json")) {
       return response.json();
     }
     throw new TypeError("Oops, we haven't got JSON!");
   })
-  .then(function(json) { /* process your JSON further */ })
-  .catch(function(error) { console.log(error); })
-  .finally(function() { isLoading = false; });
+  .then((json) => {
+    /* process your JSON further */
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:75]:
```suggestion
    /* 进一步处理 JSON */
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 8]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+  })
+  .catch((error) => {
+    console.error(error); // this line can also throw, e.g. when console = {}
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:78]:
```suggestion
    console.error(error); // 这行代码也可能抛出错误，例如：when console = {}
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 9]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+返回等效的 {{jsxref("Promise")}}。如果处理程序抛出错误或返回被拒绝的 promise，那么 `finally()` 返回的 promise 将以该值被拒绝。否则，处理程序的返回值不会影响原始 promise 的状态。
 
 ## 描述
 
-如果你想在 promise 执行完毕后无论其结果怎样都做一些处理或清理时，`finally()` 方法可能是有用的。
+如果你想在 promise 敲定时进行一些处理或者清理，无论其结果如何，那么 `finally()` 方法会很有用。
+
+`finally()` 方法类似于调用 {{jsxref("Promise/then", "then(onFinally, onFinally)")}}。然而，有几个不同之处：
+
+- 创建内联函数时，你可以将其传入一次，而不是强制声明两次或为其创建变量。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:35]:
```suggestion
- 创建内联函数时，你可以只将其传入一次，而不是强制声明两次或为其创建变量。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 10]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
+- `onFinally` 回调函数不接收任何参数。这种情况恰好适用于你*不关心*拒绝原因或兑现值的情况，因此无需提供它。
+- `finally()` 调用通常是透明的，不会更改原始 promise 的状态。例如：
+  - 与 `Promise.resolve(2).then(() => 77, () => {})` 不同，它返回一个最终会兑现为值 `77` 的 promise，而 `Promise.resolve(2).finally(() => 77)` 返回一个最终兑现为值 `2` 的 promise。
+  - 类似地，与 `Promise.reject(3).then(() => {}, () => 88)` 不同，它返回一个最终兑现为值 `88` 的 promise，而 `Promise.reject(3).finally(() => 88)` 返回一个最终以原因 `3` 拒绝的 promise。
+
+> **备注：** 在 `finally` 回调函数中抛出错误（或返回被拒绝的 promise）仍然拒绝返回的 promise。例如，`Promise.reject(3).finally(() => { throw 99; })` 和 `Promise.reject(3).finally(() => Promise.reject(99))` 都以理由 `99` 拒绝返回的 promise。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:41]:
```suggestion
> **备注：** 在 `finally` 回调函数中抛出错误（或返回被拒绝的 promise）仍会导致返回的 promise 被拒绝。例如，`Promise.reject(3).finally(() => { throw 99; })` 和 `Promise.reject(3).finally(() => Promise.reject(99))` 都以理由 `99` 拒绝返回的 promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md [hunk 11]
```diff
@@ -3,51 +3,83 @@ title: Promise.prototype.finally()
...
 
-`finally()` 虽然与 `.then(onFinally, onFinally)` 类似，它们不同的是：
+与 {{jsxref("Promise/catch", "catch()")}} 一样，`finally()` 在内部调用其调用对象上的 `then` 方法。如果 `onFinally` 不是函数，则调用 `then()` 时使用 `onFinally` 同时作为两个参数——对于 {{jsxref("Promise.prototype.then()")}}，这意味着没有附加有效的处理程序。否则，`then()` 被调用时会使用两个内部创建的函数，其行为如下：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/promise/finally/index.md:43]:
```suggestion
与 {{jsxref("Promise/catch", "catch()")}} 一样，`finally()` 在内部调用其调用对象上的 `then` 方法。如果 `onFinally` 不是函数，则调用 `then()` 时使用 `onFinally` 同时作为两个参数——对于 {{jsxref("Promise.prototype.then()")}}，这意味着没有附加有效的处理器。否则，`then()` 被调用时会使用两个内部创建的函数，其行为如下：
```

**Review by yin1999 — COMMENTED** [review]
Thanks @robert-jx. Just some suggestions

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**